logstash-core/src/test/java/org/logstash/common/io/MemoryCheckpointTest.java

```diff
@@ -39,7 +41,11 @@ public void writeNewReadExisting() throws IOException {
 
     @Test
     public void readInnexisting() throws IOException {
-        Checkpoint checkpoint = io.read("checkpoint.invalid");
-        assertThat(checkpoint, is(equalTo(null)));
+        try {
+            io.read("checkpoint.invalid");
+            fail("expected NoSuchFileException");
+        } catch (NoSuchFileException e) {
```

Comment:
Testing suggestion:

* Change annotation to: `@Test(exception = NoSuchFileException)` (or whatever, I'm writing by hand here, not tested)
* Change function to be 1 line: `io.read("checkpoint.invalid")`